From 7ef664f614d02a27720edfc851bffa3d14386c10 Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Thu, 5 Nov 2015 14:36:12 -0800
Subject: [PATCH 0970/2848] HDFS-9377. Fix findbugs warnings in
 FSDirSnapshotOp. (Mingliang Liu via Yongjun
 Zhang)

(cherry picked from commit 286cc6483fe458894cdd22ff1d8822c304e0d94f)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirSnapshotOp.java

Change-Id: I2efb510325fe30581bcabe6dfcd8c086a3ae6ee5
---
 .../hadoop/hdfs/server/namenode/FSDirectory.java   |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
index 8b24848..1e41362 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java
@@ -1551,8 +1551,7 @@ private DirectoryListing getSnapshotsListing(String src, byte[] startAfter)
   Collection<String> getSnapshotFiles(
       List<DirectorySnapshottableFeature> lsf,
       String file) throws IOException {
-    ArrayList<String> snaps = new ArrayList<String>();
-    ListIterator<DirectorySnapshottableFeature> sfi = lsf.listIterator();
+    ArrayList<String> snaps = new ArrayList<>();
     for (DirectorySnapshottableFeature sf : lsf) {
       // for each snapshottable dir e.g. /dir1, /dir2
       final ReadOnlyList<Snapshot> lsnap = sf.getSnapshotList();
-- 
1.7.9.5

